### ISSUE TITLE
Invalid Type Suggestion

### ISSUE THREAD

**Issue by Shapedsundew9** [microsoft/pylance-release#3824]
## Issue
Intellisense is suggesting the following edit but it makes no sense (use before definition)
![image](https://user-images.githubusercontent.com/16618209/211944173-a14cdbf5-baa4-4a19-a1f5-65d0ed74b340.png)
Accepting the edit results in:
![image](https://user-images.githubusercontent.com/16618209/211944805-fb0482a4-2321-4498-bbd9-c01c0cb7f4b9.png)

## Environment data
Pylance language server 2023.1.10 (pyright bbf0ae78)

Ubuntu 22.04.1 LTS

VS Code
Version: 1.74.3
Commit: 97dec172d3256f8ca4bfb2143f3f76b503ca0534
Date: 2023-01-09T16:57:40.428Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Linux x64 5.15.0-57-generic
Sandboxed: No

Python 3.11.1

## Code Snippet
```python
from typing import NewType

Conversions = NewType('Conversions', int)
```

## Expected behavior
Type definition not proposed (or something valid)

## Actual behavior
Suggestion is invalid.

## Logs
```
[Info  - 6:39:36 p.m.] (46461) Pylance language server 2023.1.10 (pyright bbf0ae78) starting
[Info  - 6:39:36 p.m.] (46461) Server root directory: /home/shapedsundew9/.vscode/extensions/ms-python.vscode-pylance-2023.1.10/dist
[Info  - 6:39:36 p.m.] (46461) Starting service instance "scratch"
[Info  - 6:39:36 p.m.] (46461) Notebook support: LSP
[Info  - 6:39:36 p.m.] (46461) Interactive window support: LSP
[Info  - 6:39:36 p.m.] (46461) No configuration file found.
[Info  - 6:39:36 p.m.] (46461) No pyproject.toml file found.
[Info  - 6:39:36 p.m.] (46461) Setting pythonPath for service "scratch": "/home/shapedsundew9/Projects/venv/bin/python3"
[Error - 6:39:36 p.m.] (46461) venvPath /home/shapedsundew9/Projects/scratch/~/Projects is not a valid directory.
[Warn  - 6:39:36 p.m.] (46461) stubPath /home/shapedsundew9/Projects/scratch/typings is not a valid directory.
[Info  - 6:39:37 p.m.] (46461) Assuming Python version 3.11
[Info  - 6:39:37 p.m.] (46461) Assuming Python platform Linux
[Info  - 6:39:37 p.m.] (46461) Searching for source files
[Info  - 6:39:37 p.m.] (46461) Found 1 source file
[Info  - 6:39:37 p.m.] (46461) Background analysis(1) root directory: /home/shapedsundew9/.vscode/extensions/ms-python.vscode-pylance-2023.1.10/dist
[Info  - 6:39:37 p.m.] (46461) Background analysis(1) started
[Info  - 6:39:38 p.m.] (46461) Indexer background runner(2) root directory: /home/shapedsundew9/.vscode/extensions/ms-python.vscode-pylance-2023.1.10/dist (index)
[Info  - 6:39:38 p.m.] (46461) Indexing(2) started
[Info  - 6:39:39 p.m.] (46461) scanned(2) 246 files over 1 exec env
[Info  - 6:39:44 p.m.] (46461) [IDX(2)] Long operation: index execution environment /home/shapedsundew9/Projects/scratch (4532ms)
[Info  - 6:39:44 p.m.] (46461) [IDX(2)] Long operation: index packages /home/shapedsundew9/Projects/scratch (4535ms)
[Info  - 6:39:44 p.m.] (46461) indexed(2) 155 files over 1 exec env
[Info  - 6:39:44 p.m.] (46461) Indexing finished(2).
```

**Comment by rchiodo** [comment]
Thanks for the issue.

I don't see that suggestion. How did you get the suggestion to show up?

![newtype](https://user-images.githubusercontent.com/19672699/211946740-ca5703c9-3117-4a00-96dc-c81787cbed3f.gif)

**Comment by rchiodo** [comment]
Do you have github copilot installed? It provides shortcuts like that, although I can't get it to behave the way you're showing either.

**Comment by rchiodo** [comment]
Thanks. I didn't have inlay hints on. Didn't realize you could double click to add them. 

I wonder if inlay hints should always use the 'string' version of a type to eliminate this sort of problem.

**Comment by Shapedsundew9** [comment]
So I closed all projects in vscode (so they would not open on startup), shut it down, rebooted my machine, opened vs code, opened and empty project (just one .py file with nothing in it) and typed:
```python
from typing import NewType, Type
C = NewType('C', int)
```
I do not even have to hit enter at the end of the second line and the suggestion pops in:
![image](https://user-images.githubusercontent.com/16618209/211950217-59e63a2d-f0d0-4e4b-b970-8017cc2193f6.png)

**Comment by Shapedsundew9** [comment]
Comment overlap. OK - so you can reproduce. Thanks.

**Comment by rchiodo** [comment]
Inlay hints should probably be showing nothing in this situation. I believe because it will always be an error to have a type annotation here.

**Comment by erictraut** [comment]
We shouldn't be offering to insert type annotations for `NewType` or similar statements. It is effectively a type declaration, so a type annotation doesn't make sense here. The same is true for statements that call `TypeVar`, `ParamSpec`, `TypeVarTuple`, `NamedTuple`, `namedtuple`, `TypedDict`, etc.

**Comment by Shapedsundew9** [comment]
Bumped into this again with EnumInt: 
![image](https://user-images.githubusercontent.com/16618209/221365315-07e53e90-d3ed-442e-851c-4eee7a069ca6.png)
Suggestion (in white, above) results in an undefined type error (below) 
![image](https://user-images.githubusercontent.com/16618209/221365403-82b1b340-57df-440a-a917-0ae084896fb5.png)

[Info  - 10:13:25] (50712) Pylance language server 2023.2.40 (pyright 085dfbfa) starting

**Comment by Shapedsundew9** [comment]
I guess a natural consequence is pylance getting upset when I do type the literal:
![image](https://user-images.githubusercontent.com/16618209/221365830-3b7e5b22-3eda-4086-ba93-d348f7abb523.png)

**Comment by erictraut** [comment]
@Shapedsundew9, you can work around the forward declaration in two ways:
1. Add a `from __future__ import annotations` to the top of your file. This tells the Python interpreter that it should defer execution of all annotations, so a forward reference will not cause a runtime exception. (This will likely be the default behavior in Python 3.12, which is still under development.)
2. Enclose the entire type annotation in quotes ("Literal[EnumerationClass.A]").
3. (best option) Omit the type annotation in this case because there's no value in annotating the type of an enum member. The type of such members are trivial for a type checker to infer.

@bschnurr, pylance really shouldn't offer to insert a type annotation for enum members in the first place.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.3.22, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023322-15-March-2023-PreRelease)
